src/metrics/metrics.ts

```diff
@@ -221,102 +225,131 @@ export function getAverageDistance(
+  const killsDeathsAssistsMap: {
+    [key: string]: { kills: number; deaths: number; assists: number };
+  } = {};
+  
+  const killDeathRatios = {}
+
+  let lastMove: PrettyMove | null = null;
 
   // look at each game and find the piece with the largest kill/death ratio
   for (const game of games) {
-    const basePieceSquares = new Map<Square, UnambiguousPieceSymbol>();
-    basePieceSquares.set('a1', 'ra');
-    basePieceSquares.set('b1', 'nb');
-    basePieceSquares.set('c1', 'bc');
-    basePieceSquares.set('d1', 'q');
-    basePieceSquares.set('e1', 'k');
-    basePieceSquares.set('f1', 'bf');
-    basePieceSquares.set('g1', 'ng');
-    basePieceSquares.set('h1', 'rh');
-    basePieceSquares.set('a2', 'pa');
-    basePieceSquares.set('b2', 'pb');
-    basePieceSquares.set('c2', 'pc');
-    basePieceSquares.set('d2', 'pd');
-    basePieceSquares.set('e2', 'pe');
-    basePieceSquares.set('f2', 'pf');
-    basePieceSquares.set('g2', 'pg');
-    basePieceSquares.set('h2', 'ph');
-    basePieceSquares.set('a8', 'RA');
-    basePieceSquares.set('b8', 'NB');
-    basePieceSquares.set('c8', 'BC');
-    basePieceSquares.set('d8', 'Q');
-    basePieceSquares.set('e8', 'K');
-    basePieceSquares.set('f8', 'BF');
-    basePieceSquares.set('g8', 'NG');
-    basePieceSquares.set('h8', 'RH');
-    basePieceSquares.set('a7', 'PA');
-    basePieceSquares.set('b7', 'PB');
-    basePieceSquares.set('c7', 'PC');
-    basePieceSquares.set('d7', 'PD');
-    basePieceSquares.set('e7', 'PE');
-    basePieceSquares.set('f7', 'PF');
-    basePieceSquares.set('g7', 'PG');
-    basePieceSquares.set('h7', 'PH');
-
-    const chess = new Chess(); // Create a new instance of the Chess class
-    chess.loadPgn(game.moves);
-    const moveHistory = chess.history();
-
-    // duplicate the base map
-    const pieceSquares = new Map<Square, UnambiguousPieceSymbol>(
-      basePieceSquares
-    );
-
-    // create an object to track kills, deaths, and assists of each piece
-    const killDeathMap: {
-      [key: string]: { kills: number; deaths: number; assists: number };
-    } = {};
-    for (const piece of pieceSquares.values()) {
-      killDeathMap[piece] = { kills: 0, deaths: 0, assists: 0 };
-    }
 
-    //
-    for (const move of moveHistory) {
-      if (move.captured) {
-        if (!kills[move.piece]) {
-          kills[move.piece] = 0;
-        }
-        kills[move.piece]++;
+    const chess = new Chess();
+    const moveGenerator = chess.historyGenerator(game.moves);
+    console.log(`game.moves: ${game.moves}`);
+    const siteLink = game.metadata[1].match(/"(.*?)"/)[1];
+    console.log(`lichess link to game played: ${siteLink}`);
+
+    for (let moveInfo of moveGenerator) {
+      const { move, board } = moveInfo;
+      //console.log("move: ", move)
+      //console.log("board: ", board)
+      lastMove = move;
```

Comment:
I don't understand why you are reassigning move here

Reply:
need to review later, can remove if unnecessary